File: core-scala/src/main/java/io/asterisque/core/wire/UnsupportedProtocolException.java
package io.asterisque.core.wire;

import io.asterisque.wire.ProtocolException;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * 新しい Wire や Server を構築するときの URI Schema が認識できない時に発生する例外です。
 */
public class UnsupportedProtocolException extends ProtocolException {
  public UnsupportedProtocolException(@Nonnull String message) {
    super(message, null);
  }

  public UnsupportedProtocolException(@Nonnull String message, @Nullable Throwable ex) {
    super(message, ex);
  }
}

File: core-scala/src/main/scala/io/asterisque/wire/gateway/UnsupportedProtocolException.scala
package io.asterisque.wire.gateway

import javax.annotation.{Nonnull, Nullable}
import io.asterisque.wire.ProtocolException

/**
  * 新しい Wire や Server を構築するときの URI Schema が認識できない時に発生する例外です。
  */
class UnsupportedProtocolException(@Nonnull message:String, @Nullable ex:Throwable) extends ProtocolException(message, ex) {
  def this(@Nonnull message:String) = this(message, null)
}
